src/cosim/timer.cpp

```diff
@@ -31,83 +30,59 @@ class real_time_timer::impl
         startTime_ = Time::now();
         rtStartTime_ = startTime_;
         rtCounter_ = 0L;
-        measuredRealTimeFactor_ = 1.0;
     }
 
     void sleep(time_point currentTime)
     {
-        Time::time_point current = Time::now();
-        update_real_time_factor(current, currentTime);
-        lastSimulationTime_ = currentTime;
-        if (realTimeSimulation_) {
-            const auto elapsed = current - startTime_;
-            const auto expectedSimulationTime = (currentTime - simulationStartTime_) / realTimeFactorTarget_.load();
+        const auto newHash = std::hash<real_time_config>()(*config_);
+        if (newHash != configHashValue_) {
+            start(currentTime);
+            configHashValue_ = newHash;
+        }
+        if (config_->real_time_simulation && config_->real_time_factor_target.load() > 0.0) {
```

Comment:
You call `config_->real_time_factor_target.load()` twice here, first in line 42 and then in line 44. But if the `real_time_config` struct is designed to be accessed by multiple threads simultaneously, then it may change between these two loads. It could therefore be `> 0.0` at first, so that the `if` block is entered, but then be `== 0.0` on the second load, triggering a divide-by-zero error.

A simple solution would be to load it to some local variable and use that instead.

Reply:
Dang! Will fix.